addons/website_sale/models/website_snippet_filter.py

```diff
@@ -167,7 +167,7 @@ def _get_products_recently_sold_with(
         self, website, limit, domain, product_template_id, **kwargs,
     ):
         products = self.env['product.product']
-        current_template = self.env['product.template'].browse(int(product_template_id)).exists()
+        current_template = product_template_id and self.env['product.template'].browse(int(product_template_id)).exists()
```

Comment:
```suggestion
        current_template = product_template_id and \
            self.env['product.template'].browse(int(product_template_id)).exists()
```
```suggestion
        product_template_id = int(product_template_id) if product_template_id else None
        current_template = self.env['product.template'].browse(product_template_id).exists()
```

Lines should be kept under 100 chars length, so either split them, or convert the `product_template_id` to a valid value on a different line.